server/room.h

```diff
@@ -0,0 +1,44 @@
+#ifndef ROOM_H
+#define ROOM_H
+
+#include <vector>
+#include <memory>
+#include <optional>
+#include "Sensors/sensor.h"
+#include "Devices/device.h"
+#include "andreea_unique_ptr"
+
+namespace smart_home {
+
+class Room : public StatusObject
+{
+public:
+    Room(const std::string &name = "");
+
+    void AddSensor(MyUniquePtr<Sensor> &&sensor);
+    bool DeleteSensor(const std::string &name);
+
+    void AddDevice(MyUniquePtr<Device> &&device);
+    bool DeleteDevice(const std::string &name);
+
+    std::string& GetName();
```

Comment:
when you return references to class members, make sure that you return them as const.